docs/enhancements/dual-stack-vips.md

```diff
@@ -0,0 +1,104 @@
+---
+title: dual-stack-vips
+authors:
+  - "@mkowalski"
+creation-date: 2022-08-07
+last-updated: 2022-08-07
+tracking-link:
+  - https://issues.redhat.com/browse/MGMT-9915
+see-also:
+  - https://github.com/openshift/enhancements/blob/master/enhancements/network/on-prem-dual-stack-vips.md
+---
+
+# Dual-stack VIPs
+
+## Summary
+
+Originally Assisted Installer supported only single-stack IPv4 or IPv6 deployments. When dual-stack support was added, we carried over the VIP configuration that only allowed for a single VIP to be specified.
+
+The initial rationale was that in dual-stack clusters every required communication can happen over IPv4 and only the workload running in the cluster may need IPv6 connectivity.
+
+Given this is no longer the case and that OCP for the baremetal platform supports dual-stack VIPs, this design is proposing the addition of a Cluster property allowing configuration of the aforementioned feature.
+
+## Motivation
+
+The current implementation of dual-stack support assumes that everything in the cluster should be reachable via IPv4 and that IPv6 is an addition for the workload running in the cluster. This assumption makes it impossible to interact with such a cluster using IPv6-only client, i.e. makes it impossible for IPv6-only Hub to manage a dual-stack Spoke.
+
+### Goals
+
+* Allow configuring both IPv4 and IPv6 VIP for dual-stack clusters.
+
+### Non-goals
+
+* Allow configuring more than single IPv4 and single IPv6 VIP. This is still not supported by the OCP.
+
+## Proposal
+
+### User Stories
+
+* As an operator of the IPv6-only Hub cluster I want to deploy dual-stack Spoke cluster.
+
+* As a deployer of a dual stack OCP cluster I want to run IPv6-only applications that need access to the API and Ingress VIPs.
+
+### API Extensions
+
+New fields will have to be added to the Cluster object to persist the specified VIPs.
+
+### Implementation Details
+
+#### Phase 1
+
+`cluster-create-params`, `v2-cluster-update-params` and `cluster` will get new fields
+
+* `api_vips` of type `array`
+* `ingress_vips` of type `array`
+
+Respective AgentClusterInstall CR will also get those new fields.
+
+The database schema will be updated by creating new tables `api_vips` and `ingress_vips` with the following columns
+
+* `cluster_id`
+* `vip`
+
+#### Phase 2
+
+A set of functions for cluster creation and cluster update will provide a compatiblity between already existing fields (`api_vip` and `ingress_vip`) and new ones (`api_vips` and `ingress_vips`). The behaviour will be as follows
```

Comment:
What about just following the [k8s API change guide](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#making-a-singular-field-plural)?

Reply:
Done